kyaml/setters2/set.go

```diff
@@ -244,8 +244,15 @@ func validateAgainstSchema(ext *CliExtension, sch *spec.Schema) error {
 
 	var inputYAML string
 	if len(ext.Setter.ListValues) > 0 {
-		tmpl, err := template.New("validator").
-			Parse(`{{.key}}:{{block "list" .values}}{{"\n"}}{{range .}}{{println "-" .}}{{end}}{{end}}`)
+		var tmplText string
+		if sch.Items != nil && sch.Items.Schema != nil &&
+			sch.Items.Schema.Type.Contains("string") {
+			tmplText = `{{.key}}:{{block "list" .values}}{{"\n"}}{{range .}}{{printf "- %q\n" .}}{{end}}{{end}}`
```

Comment:
Can you add a reference document on the tmplText as to what it means and how it's parsed.